reactive/sensu-client.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+# Copyright (C) 2017  Qrama, developed by Tengu-team
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# pylint: disable=c0111,c0103,c0301
+import os
+import shutil
+from subprocess import call, CalledProcessError
+from charmhelpers.core.templating import render
+from charmhelpers.core.hookenv import local_unit, status_set, config, open_port, application_version_set, unit_public_ip
+from charmhelpers.core.host import service_restart
+from charms.reactive import when, when_not, set_state, remove_state
+
+
+CONFIG_DIR = '/etc/sensu/conf.d'
+SSL_DIR = '/etc/sensu/ssl'
+
+
+@when('apt.installed.sensu', 'info.available')
+@when_not('sensu.installed')
+def setup_sensu(info):
+    application_version_set('0.29')
+    if not os.path.isdir(SSL_DIR):
+        os.mkdir(SSL_DIR)
+    with open('{}/ssl_key.pem'.format(SSL_DIR), 'w+') as ssl_key:
+        ssl_key.write(config()['ssl_key'])
+    with open('{}/ssl_cert.pem'.format(SSL_DIR), 'w+') as ssl_cert:
+        ssl_cert.write(config()['ssl_cert'])
+    rabbitmq = {'host': config()['rabbitmq'].split(':')[0],
```

Comment:
It's more user-friendly to do some basic checking of config values and go into a blocked state if they aren't correct. Currently, the charm code will fail, putting the charm into an error state.

This is ofcourse only relevant if these config options are directed towards users. If Tengu itself will fill in these config parameters, you shouldn't spend time on doing this.